Library/Homebrew/test/ARGV_test.rb

```diff
@@ -1,79 +1,147 @@
 require "testing_env"
 require "extend/ARGV"
 
-class ArgvExtensionTests < Homebrew::TestCase
-  def setup
-    super
-    @argv = [].extend(HomebrewArgvExtension)
-  end
+describe HomebrewArgvExtension do
```

Comment:
I don't understand the motivation for rewriting this test. How does this make things better for our users?